os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -206,35 +191,161 @@ binmgr_uinfo_t *binary_manager_get_udata(uint32_t bin_idx)
-	g_bin_count++;
-	BIN_ID(g_bin_count) = -1;
-	BIN_FILECNT(g_bin_count) = 1;
-	BIN_RTLIST(g_bin_count) = NULL;
-	BIN_NRTLIST(g_bin_count) = NULL;
-	BIN_STATE(g_bin_count) = BINARY_INACTIVE;
-	BIN_VER(g_bin_count, 0) = version;
-	BIN_LOAD_PRIORITY(g_bin_count, 0) = load_priority;
-	strncpy(BIN_NAME(g_bin_count), name, BIN_NAME_MAX - 1);
-	BIN_NAME(g_bin_count)[BIN_NAME_MAX - 1] = '\0';
-	sq_init(&BIN_CBLIST(g_bin_count));
-
-	bmvdbg("[USER %d] %s\n", g_bin_count, BIN_NAME(g_bin_count));
+	BIN_ID(bin_idx) = -1;
+	BIN_RTLIST(bin_idx) = NULL;
+	BIN_NRTLIST(bin_idx) = NULL;
+	BIN_STATE(bin_idx) = BINARY_INACTIVE;
+	BIN_PARTNUM(bin_idx, 0) = part_num;
+	BIN_PARTSIZE(bin_idx, 0) = part_size;
+	strncpy(BIN_NAME(bin_idx), name, BIN_NAME_MAX - 1);
+	BIN_NAME(bin_idx)[BIN_NAME_MAX - 1] = '\0';
+	sq_init(&BIN_CBLIST(bin_idx));
 
-	return g_bin_count;
+	bmvdbg("[USER%d : 1] %s size %d num %d\n", bin_idx, BIN_NAME(bin_idx), BIN_PARTSIZE(bin_idx, 0), BIN_PARTNUM(bin_idx, 0));
+}
+
+/****************************************************************************
+ * Name: binary_manager_scan_ubin_all
+ *
+ * Description:
+ *	 This function scans user binaries and update information based on bootparam.
+ *
+ ****************************************************************************/
+bool binary_manager_scan_ubin_all(void)
+{
+	int ret;
+	int bin_idx;
+	int part_idx;
+	int bp_app_idx;
+	bool is_found;
+	uint32_t version;
+	binmgr_bpdata_t *bp_data;
+	common_binary_header_t common_header_data;
+	user_binary_header_t user_header_data;
+	char devpath[BINARY_PATH_LEN];
+
+	is_found = false;
+
+	bp_data = binary_manager_get_bpdata();
+	/* Update user binary data based on bootparam */
+	for (bp_app_idx = 0; bp_app_idx < bp_data->app_count; bp_app_idx++) {
+		bin_idx = binary_manager_get_index_with_name(bp_data->app_data[bp_app_idx].name);
+		if (bin_idx < 0) {
+			bmdbg("Failed to find matched binary %s in binary table \n", bp_data->app_data[bp_app_idx].name);
+			continue;
+		}
+		BIN_BPIDX(bin_idx) = bp_app_idx;
+		part_idx = bp_data->app_data[bp_app_idx].useidx;
+		snprintf(devpath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, BIN_PARTNUM(bin_idx, part_idx));
+		if (bin_idx == BM_CMNLIB_IDX) {
+			ret = binary_manager_read_header(BINARY_COMMON, devpath, (void *)&common_header_data, false);			
+			version = common_header_data.version;
+		} else {
+			ret = binary_manager_read_header(BINARY_USERAPP, devpath, (void *)&user_header_data, false);			
+			version = user_header_data.bin_ver;
+		}
+		if (ret == OK) {
+			/* Return true it there is at least one valid binary */
+			BIN_USEIDX(bin_idx) = part_idx;
+			BIN_COUNT(bin_idx) = 1;
+			BIN_VER(bin_idx, part_idx) = version;			
+			bmvdbg("[%d] part index %d, version %d\n", bin_idx, part_idx, version);
+			if (bin_idx != BM_CMNLIB_IDX) {
+				BIN_LOAD_PRIORITY(bin_idx, part_idx) = user_header_data.loading_priority;
+			}
+			/* Return true it there is at least one valid binary */
```

Comment:
`Return true it there is at least one valid binary` -> `Return true if there is at least one valid binary`